### DISCUSSION TITLE
H.264, H.265, etc. benchmarks?

### DISCUSSION THREAD

**Discussion by dagoldman** [Breakthrough/DVR-Scan#69] [General]
I think it might be helpful to have benchmarks related to the input video compression standard. I believe it would be helpful for me.  I am tempted to just use H.265+ but if I knew dvr-scan processed one minute of H.264 a lot faster than one minute of H.265+ I might use H.264 instead.

Most technologies have pros / cons. H.265 compresses a lot more than H.264. Ditto for H.265+ vs H.265. That’s an advantage. But is it possible dvr-scan is much slower to process a minute of H.265 than a minute of H.264? Are there any tests answering the question?

If not already dvr-scan benchmarks comparing H.264 / H.264+ / H.265 / H.265+, I would suggest something like the following: For each standard, make a typical surveillance video file. Tabulate file size, minutes of video, time for dvr-scan to process file, output avi file size.

I understand each input video file is somewhat different. Different frames per second. Different scene. Different activity. So perhaps no “typical” file. And dvr-scan run time depends on the cpu. But I think a comparison might still be useful, because we don’t care so much about a 10% difference or something like that. Instead, we want a much larger performance difference, such as 100%. The benchmark we are looking for, I believe, is: # of seconds dvr-scan takes to process one minute (or other interval) of video 

I may be totally off. I am ready to be corrected. But in general, benchmarking is helpful for many purposes. And especially for this kind of long-running application.

Daniel

**Comment by Breakthrough** [comment]
While this is technically possible, the scope of this is pretty big to be honest, and depends quite a bit on your particular system, operating system, exact version of libraries being used, etc.  I'm not sure I know enough about all the variables involved to back any claims like that in general.

I agree such a comparison is very useful, but I think you could perform such a benchmark for your own use case, with your own hardware, and get much more accurate and relevant numbers.  DVR-Scan already outputs the average processing speed in FPS, so you could just run it a few times on various types of videos to see if any work particularly better.

If you or anyone else do end up running such tests please, do post the results here along with your system specs/library versions.
